galsim/wfirst/wfirst_psfs.py

```diff
@@ -0,0 +1,460 @@
+#
+# This file is part of GalSim: The modular galaxy image simulation toolkit.
+# https://github.com/GalSim-developers/GalSim
+#
+# GalSim is free software: redistribution and use in source and binary forms,
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions, and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions, and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+#
+
+import galsim
+import galsim.wfirst
+import numpy as np
+import os
+import pyfits
+
+"""
+@file wfirst_psfs.py
+
+Part of the WFIRST module.  This file includes routines needed to define a realistic PSF for WFIRST.
+"""
+
+# Define a default set of bandpasses for which this routine works.
+default_bandpass_list = ['J129', 'F184', 'W149', 'Y106', 'Z087', 'H158']
+# Prefix for files containing information about Zernikes for each SCA.
+zemax_filepref = os.path.join(galsim.meta_data.share_dir,
+                              "AFTA_WFI_v4-2-5_140326_192nmRMS_NoAps_PLT_Zemax_ZernStanTerm_C")
+zemax_filesuff = '_F01_W04.txt'
+zemax_wavelength = 1293. #nm
+
+def getPSF(SCAs=None, approximate_struts=False, n_waves=None, extra_aberrations=None,
+           wavelength_limits=None, verbose=True, achromatic=False):
+    """
+    Get the PSF for WFIRST observations.
+
+    By default, this routine returns a list of ChromaticOpticalPSF objects, with the list index
+    corresponding to the SCA (Sensor Chip Array, the equivalent of a chip in an optical CCD).  The
+    PSF corresponds to a location in the center of the SCA.  Currently we do not have information
+    about PSF variation across the SCAs, which is expected to be relatively small.
+
+    This routine also takes an optional keyword `SCAs`, which can be a single value or a list; if
+    this is specified then results are not included for the other SCAs.  However, to preserve the
+    indexing, the list of results has the same length, with the result being None for the SCAs that
+    are not of interest.
+
+    The default is to do the calculations using the full specification of the WFIRST pupil plane,
+    which is a costly calculation in terms of memory.  To turn this off, use the optional keyword
+    `approximate_struts`.  In this case, the pupil plane will have the correct obscuration and
+    number of struts, but the struts will be purely radial and evenly spaced instead of the true
+    configuration.  The simplicity of this arrangement leads to a much faster calculation, and
+    somewhat simplifies the configuration of the diffraction spikes.  Also note that currently the
+    orientation of the struts is fixed, rather than rotating depending on the orientation of the
+    focal plane.
```

Comment:
Perhaps mention that the rotation is easily effected by the user if they care about that:
```
psf = galsim.wfirst.getPSF(...).rotate(angle)
```